### PR TITLE
docs: Add upgrade guide for Algolia plugin

### PR DESCRIPTION
**Pull request by olivermrbl**
**What**
Upgrade guide for Algolia serach plugin

Closes DOCS-665

### CODE REVIEW

**Comment by olivermrbl** [comment]
Exact version still to be decided. Please keep as draft until then :)

**Comment by olivermrbl** [comment]
@shahednasser first time I see this:
CleanShot 2023-03-27 at 09 39 11@2x


That is really nifty! 💯

File: docs/content/upgrade-guides/plugins/algolia/1-0-0.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Algolia Search: v1.0.0
+
+Version 1.0.0 of our official Algolia plugin, `medusa-plugin-algolia` comes with breaking changes to the plugin options that are passed to the Algolia service through `medusa-config.js`.
```
Comment (shahednasser) [inline docs/content/upgrade-guides/plugins/algolia/1-0-0.md:3]:
```suggestion
Version 1.0.0 of the official Algolia plugin comes with breaking changes to the plugin options that are passed to the Algolia service through `medusa-config.js`.
```

File: docs/content/upgrade-guides/plugins/algolia/1-0-0.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      indexSettings: {
+        searchableAttributes: string[]
+        attributesToRetrieve: string[]
+				...
+      }
+      transformer?: <T>(medusaEntity: T) => unknown
+    }
+  }
+}
+```
```
Comment (shahednasser) [inline docs/content/upgrade-guides/plugins/algolia/1-0-0.md:26]:
~~~suggestion
```js title=medusa-config.js
const plugins = [
  // ...
  {
    applicationId: string
    adminApiKey: string
    settings: {
      indexName: {
        indexSettings: {
          searchableAttributes: string[]
          attributesToRetrieve: string[]
          // ...
        }
        transformer?: (product) => ({
          id: product.id,
          // ...
        })
      }
    }
  }
]
```
~~~

File: docs/content/upgrade-guides/plugins/algolia/1-0-0.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  admin_api_key: "someApiKey",
+  settings: {
+    // example
+    products: {
+      searchableAttributes: ["title", "description"],
+      attributesToRetrieve: ["title", "description"],
+		}
+  }
+}
+```
```
Comment (shahednasser) [inline docs/content/upgrade-guides/plugins/algolia/1-0-0.md:46]:
~~~suggestion
```js title=medusa-config.js
const plugins = [
  // ...
  {
    application_id: "someId",
    admin_api_key: "someApiKey",
    settings: {
      // example
      products: {
        searchableAttributes: ["title", "description"],
        attributesToRetrieve: ["title", "description"],
      }
    }
  }
]
```
~~~

File: docs/content/upgrade-guides/plugins/algolia/1-0-0.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+In the above example, an index `products` has been configured with two options. Updating to 1.0.0 requires you to nest these within the `indexSettings`. Additionally, we recommend updating your admin API key and application ID to use camel case, as the snake-cased version will be deprecated.
```
Comment (shahednasser) [inline docs/content/upgrade-guides/plugins/algolia/1-0-0.md:48]:
```suggestion
In the above example, an index `products` has been configured with two options `searchableAttributes` and `attributesToRetrieve`. Updating to v1.0.0 requires you to nest these within the `indexSettings`. Additionally, the admin API key and application ID options should now be in camel case, as the snake-cased version will be deprecated. 
```

File: docs/content/upgrade-guides/plugins/algolia/1-0-0.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  settings: {
+    products: {
+      indexSettings: {
+        searchableAttributes: ["title", "description"],
+        attributesToRetrieve: ["title", "description"],
+      }
+    }
+  }
+}
+```
```
Comment (shahednasser) [inline docs/content/upgrade-guides/plugins/algolia/1-0-0.md:65]:
~~~suggestion
```js title=medusa-config.js
const plugins = [
  // ...
  {
    applicationId: "someId",
    adminApiKey: "someApiKey",
    settings: {
      products: {
        indexSettings: {
          searchableAttributes: ["title", "description"],
          attributesToRetrieve: ["title", "description"],
        }
      }
    }
  }
]
```
~~~

File: docs/content/upgrade-guides/plugins/algolia/1-0-0.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+You can read more about the new plugin options properties in the package README.
```
Comment (shahednasser) [inline docs/content/upgrade-guides/plugins/algolia/1-0-0.md:67]:
```suggestion
You can learn more about the new plugin options in the [Algolia plugin documentation](../../../plugins/search/algolia.md).
```

**Review by shahednasser — CHANGES_REQUESTED** [review]
Similar suggestions to the MeiliSearch plugin. Would be great to also run the eslint command to fix eslint errors 😇

**Review by shahednasser — APPROVED** [review]
LGTM 🚀

**Pull request merged**